packages/terra-form-select/src/native-select/NativeSelect.jsx

```diff
@@ -0,0 +1,247 @@
+  /**
+   * Placeholder data. The presence of this data object is used as the indicator for whether or not a placeholder is used.
+   */
+  placeholder: PropTypes.shape({
+    /**
+     * Whether the placeholder is included as an option to clear selection.
+     */
+    allowClear: PropTypes.bool,
+    /**
+     * Optional override of the default display.
+     */
+    display: PropTypes.string,
+    /**
+     * Optional override of the default value.
+     */
+    value: PropTypes.string,
+  }),
+  /**
+   * Whether the field is required.
+   */
+  required: PropTypes.bool,
+  /**
+   * The value of the select. Can be a string or number.
+   */
+  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+};
+
+const defaultProps = {
+  disabled: false,
+  isIncomplete: false,
+  isInvalid: false,
+  options: [],
+  required: false,
+};
+
+const createPlaceholder = (placeholder, intl) => {
+  if (!placeholder) {
+    return undefined;
+  }
+  const display = placeholder.display ? placeholder.display : intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
+  const value = placeholder.value ? placeholder.value : defaultPlaceholderValue;
+  const attrs = placeholder.allowClear ? {} : { disabled: true, hidden: true };
+  return <option value={value} {...attrs}>{display}</option>;
+};
+
+const createOptions = options => {
+  const currentOptGroupKeys = [];
+
+  return options.map(current => {
+    if (current.options) {
+      const optGroupKeyIndex = geOptGroupKeyIndex(current.display, currentOptGroupKeys);
+      const optGroupKey = getOptGroupKey(current.display, optGroupKeyIndex);
+      currentOptGroupKeys.push(optGroupKey);
+
+      return (
+        <optgroup key={optGroupKey} label={current.display}>
+          {createOptions(current.options)}
+        </optgroup>
+      );
+    }
+    return <option key={`${current.value}`} value={current.value}>{current.display}</option>;
+  });
+};
+
+const NativeSelect = ({
+  ariaDescribedBy,
+  ariaLabel,
+  attrs,
+  disabled,
+  defaultValue,
+  id,
+  intl,
+  isInvalid,
+  isIncomplete,
+  onChange,
+  options,
+  placeholder,
+  required,
+  value,
+  ...customProps
+}) => {
+  const [uncontrolledValue, setUncontrolledValue] = useState(!value ? defaultValue || getFirstValue(options, placeholder) : undefined);
+  const refIsControlled = useRef(!uncontrolledValue);
+  const refSelect = useRef();
+  const theme = React.useContext(ThemeContext);
+
+  const handleOnMouseDown = () => {
+    refSelect.current.setAttribute('data-focus-interaction', 'mouse');
+  };
+  const handleOnBlur = () => {
+    refSelect.current.setAttribute('data-focus-interaction', 'none');
+  };
+  const handleOnFocus = () => {
+    if (refSelect.current.getAttribute('data-focus-interaction') !== 'mouse') {
+      refSelect.current.setAttribute('data-focus-interaction', 'keyboard');
+    }
+  };
+  const handleOnChange = event => {
+    if (onChange) {
+      onChange(event);
```

Comment:
Either the return value here needs to change or the onChange documentation in the propTypes needs to change. The documentation currently states the value will be return:

> Callback function triggered when the select value changes. function(value)

Reply:
I will update the doc.

Reply:
Updated.